src/Solvers/Greedy.jl

```diff
@@ -172,6 +178,55 @@ function greedy_choose_simple!(queue, remaining)
     return node
 end
 
+
+function greedy_choose_thermal!(queue, remaining, nbranch=8, temperature=1, rel_temperature=true)
+    n = 0
+    choices = Vector{HeapNode}()
+
+    while (!isempty(queue) && n < nbranch)
+        node = pop!(queue)
+        a, b, c = meta(node)
+
+        any(inds -> inds ∉ values(remaining), (a, b)) && continue
+
+        push!(choices, node)
+        n += 1
+    end
+
+    if n == 0
+        return nothing
+    elseif n == 1
+        return choices[1]
+    end
+
+    costs = [choice.cost for choice in choices]
+    cmin = costs[1]
+
+    rel_temperature && (temperature *= max(1, abs(cmin)))
+
+    if temperature == 0.
+        energies = [c == cmin ? 1 : 0 for c in costs]
+    else
+        energies = [exp(-(c - cmin) / temperature) for c in costs]
+    end
+
+    normalize!(energies, 1)
+
+    # choose randomly a contraction weighted by the energies
+    chosen = rand(Categorical(energies))
+
+    node = popat!(choices, chosen)
+
+
+    # put the other choice back in the heap
+    queue = BinaryMinHeap{HeapNode{Float64,NTuple{3,Set{Symbol}}}}()
```

Comment:
What do you want to do with this line? Empty the queue? If so, this doesn't take effect.

Reply:
So currently this does nothing?

Reply:
Exactly, this does nothing. Do you want to empty the queue?

Reply:
I removed it in the last commit.